### DISCUSSION TITLE
AI Assistant Configuration - API Keys - don't save or load

### DISCUSSION THREAD

**Discussion by cgraham149** [zed-industries/zed#18211] [Feature Requests]
I'm using the Zed Flatpak.

I added my Anthropic and OpenAI keys in the configuration panel but they do not save. They are only good for that session but if I restart, I have to enter the keys all over again.

I also tried adding the env to settings.json and restarted Zed but they don't load and I am given the "LLM provider not configured" warning and "Open configuration" link.

I added this to the bottomof my settings.json file:
```
"env": {
    "OPENAI_API_KEY": "my-api-key",
    "ANTHROPIC_API_KEY": "my-api-key"
  }
```
I must be doing something wrong but I don't know what it is!

**Comment by cgraham149** [comment]
1. I added the env variable for the api keys so they are available to the sandboxed flatpak - confirmed
2. I also added api_keys to settings.json as instructed, saved file, restarted Zed.

Api keys are not available in the Zed Assistant Panel Configuration tab and the warning message still states: "**LLM provider not configured**"

Can anyone help or suggest a solution or work around?

**Comment by notpeter** [comment]
See: https://github.com/zed-industries/zed/issues/18257

Install a keyring (e.g. gnome-keyring) or set the environment vars before starting Zed.

**Reply by cgraham149** [reply]
It was easier than that. It was USER error. All I had to do was hit the **ENTER** key after adding the API Key in the  Assistant Configuration panel. I was the idiot here lol!

**Reply by danielgormly** [reply]
So I didn't have gnome-keyring installed and ended up here, which is great and helpful, so thank you. But for a complete description of behaviour on Linux 6.12.3-arch1-1 x86_64, you hit enter after typing the key and nothing happens.

**Reply by autumn-ma** [reply]
Can confirm the enter key doesn't save the key, I am on manjaro.

**Reply by jd-apprentice** [reply]
Thanks for the clarification, this was clearly the issue, I've installed `gnome-keyring` in arch and it worked like a charm :)

**Reply by NicholasDewberryOfficial** [reply]
This helped me as well. Thanks!
